### DISCUSSION TITLE
JPEG 2000 support missing on Ubuntu

### DISCUSSION THREAD

**Discussion by richard-viney** [libvips/libvips#2745] [Q&A] [answered]
Hi,

Thanks so much for this wonderful library!

When using `libvips` from Homebrew on macOS, running `vips -l | grep 2k` gives output and JPEG 2000 support is present and working.

But after installing `libvips` on Ubuntu 20.04 (also tried on 21.10), running `vips -l | grep 2k` gives no output and JPEG 2000 support is missing. I've tried installing `imagemagick` and `libopenslide0` packages as well, but still no luck. Interestingly, the `openjp2` package gets installed, and is a dynamic dependency of `/usr/bin/vips`:

```bash
$ ldd /usr/bin/vips | grep jp2
	libopenjp2.so.7 => /lib/aarch64-linux-gnu/libopenjp2.so.7 (0x0000ffffb7abb000)
```

Is it possible to have JPEG 2000 support in libvips on Ubuntu? Some kind of extra step to enable it perhaps? I've seen posts regarding security issues in `openjp2`, so am wondering if it's disabled because of that?

Many thanks,

**Comment by jcupitt** [comment]
Hi @richard-viney,

jp2k support is in libvips is fairly recent (8.11), so it's not filtered through to the official ubuntu repositories yet. Perhaps 22.04 will have it included.

In the meantime, you can build yourself from source, it's not too tricky.

**Comment by richard-viney** [comment, marked as answer]
Thanks, that makes sense.

I can confirm that libvips 8.12.1 is going to be in Ubuntu 22.04, which is only a few weeks away from official release. There are already Docker images out for it, and the JPEG 2000 support looks to be working fine there.

Building libvips from source on 20.04 is slightly tricky because 20.04 packages openjpeg 2.3.1 but libvips wants at least 2.4.0.

In my case I'm going to use Ubuntu 22.04, but hopefully the above information is useful to someone else at some point.
